### PR TITLE
seed: Distinguish project updates

### PR DESCRIPTION
**Pull request by cloudhead**

### CODE REVIEW

File: seed/src/event.rs [hunk 1]
```diff
@@ -36,6 +41,15 @@ pub enum Event {
     PeerDisconnected(PeerId),
     /// A project has been tracked from a peer.
     ProjectTracked(Project, PeerId),
+    /// A project has been updated from a peer.
+    ProjectUpdated {
+        /// The URN of the updated project, including the updated path.
+        urn: RadUrn,
+        /// The updated rev, if any.
+        rev: Option<Rev>,
```
Comment (cloudhead) [inline seed/src/event.rs:49]: Not sure in what cases this is `None` :thinking:

**Review by xla — APPROVED** [review]
🆔 💉 ❕ 🐻

File: seed/src/event.rs (hunk 1 above)
Comment (FintanH) [reply seed/src/event.rs:49]: If it's `None` then it's basically just advertising the project. I'm not even sure what the protocol would do in that case. It'll possibly just mark it as uninteresting

File: seed/src/event.rs [hunk 2]
```diff
@@ -53,6 +67,18 @@ impl Event {
         })
     }
 
+    pub(crate) async fn project_updated(
```
Comment (FintanH) [inline seed/src/event.rs:70]: Why is this `async`? Why is this `Result`? WHY ARE YOU LYING?!

File: seed/src/lib.rs [hunk 3]
```diff
@@ -291,24 +294,31 @@ impl Node {
 
         let result = {
             let urn = project_urn.clone();
-            api.with_storage(move |storage| -> Result<(), librad::git::storage::Error> {
-                // FIXME(xla): There should be a saner way to test.
-                let exists = storage.has_urn(&urn)?;
+            api.with_storage(
+                move |storage| -> Result<bool, librad::git::storage::Error> {
```
Comment (FintanH) [inline seed/src/lib.rs:298]: Oooh I didn't know about this syntax :o

**Comment by kim** [comment]
Stale?

**Comment by xla** [comment]
> Stale?

Yeah, not relevant anymore afaiu. We can reopen if we need it.

**Pull request closed without merging**